src/main/java/jenkins/branch/Branch.java

```diff
@@ -94,6 +95,7 @@ public String getSourceId() {
      * @return the name of the branch.
      */
     public String getName() {
+        // TODO this could include a uniquifying prefix defined in BranchSource
```

Comment:
Well, that is handled by a name prefix given by the `SCMSource`. The idea here is a _user_-defined prefix that would enable you to add multiple sources with otherwise clashing names, potentially giving them distinct branch properties. TBD whether this is really necessary; @stephenc and I had an example in mind but on reflection this feature would neither be necessary nor usable in that scenario.

Anyway I was just recording a note that even though `BranchProjectFactory.newInstance` is now constrained to honor `Branch.name`—which the existing implementations already did, except `YamlProjectFactory` which is going to need revision anyway to provide a unified fix for JENKINS-30744—we are not prevented from having project names include prefixes, since there is no requirement that `Branch.name` equals `SCMHead.name`.